src/unstract/sdk/helper.py

```diff
@@ -16,3 +19,24 @@ def get_platform_base_url(platform_host: str, platform_port: str) -> str:
         if platform_host[-1] == "/":
             return f"{platform_host[:-1]}:{platform_port}"
         return f"{platform_host}:{platform_port}"
+
+    @staticmethod
+    def is_public_adapter(adapter_id: str) -> bool:
+        """Check if the given adapter_id is one of the public adapter keys.
+
+        This method iterates over the attributes of the PublicAdapterKeys class
+        and checks if the provided adapter_id matches any of the attribute values.
+
+        Args:
+            adapter_id (str): The ID of the adapter to check.
+
+        Returns:
+            bool: True if the adapter_id matches any public adapter key, False otherwise.
+        """
+        try:
+            # Retrieve all attribute names from the PublicAdapterKeys class
+            attributes = dir(PublicAdapterKeys)
+            public_adapter_keys = {getattr(PublicAdapterKeys, attr) for attr in attributes}
+            return adapter_id in public_adapter_keys
+        except Exception:
+            return False
```

Comment:
NIT: You could log any errors here as a warning

Reply:
@tahierhussain resolving this - please add it later if you feel its necessary

Reply:
@chandrasekharan-zipstack Done. The warning log has been added.